### PR TITLE
Initial implementation of invocation features

### PR DESCRIPTION
**Pull request by fabiocav**
This PR introduces a way to provide invocation feature services that will extent the capabilities of the invocation context.

### CODE REVIEW

File: src/DotNetWorker/Context/Features/ConverterBidingFeature.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    internal class ConverterBidingFeature
```
Comment (brettsam) [inline src/DotNetWorker/Context/Features/ConverterBidingFeature.cs:6]: Spelling?

File: src/DotNetWorker/Configuration/ServiceCollectionExtensions.cs [hunk 2]
```diff
@@ -56,6 +57,10 @@ public static IFunctionsWorkerApplicationBuilder AddFunctionsWorker(this IServic
             // Function Execution Contexts
             services.AddSingleton<IFunctionContextFactory, DefaultFunctionContextFactory>();
 
+            // Invocation Features
+            services.TryAddSingleton<IInvocationFeaturesFactory, DefaultInvocationFeaturesFactory>();
+            services.AddSingleton<IInvocationFeatureProvider, ConverterBindingFeatureProvider>();
```
Comment (brettsam) [inline src/DotNetWorker/Configuration/ServiceCollectionExtensions.cs:62]: Would you expect we move stuff like the FunctionInvocation off of the context directly, and into a Feature?

File: src/DotNetWorker/Context/Features/ConverterBidingFeature.cs (hunk 1 above)
Comment (fabiocav) [reply src/DotNetWorker/Context/Features/ConverterBidingFeature.cs:6]: Should have left this type out.... was mostly stubbing this. I'll actually delete this class from this change

File: src/DotNetWorker/Configuration/ServiceCollectionExtensions.cs (hunk 2 above)
Comment (fabiocav) [reply src/DotNetWorker/Configuration/ServiceCollectionExtensions.cs:62]: Perhaps the core logic. Still having the middleware, but potentially moving to this.

**Review by brettsam — APPROVED** [review]

**Pull request merged**